dpp/src/state_transition/validation/validator_state_transition_identity_signature.rs

```diff
@@ -42,26 +42,56 @@ pub async fn validate_state_transition_identity_signature(
         && public_key.get_type() != KeyType::ECDSA_HASH160
     {
         validation_result.add_error(SignatureError::InvalidIdentityPublicKeyTypeError {
-            key_type: public_key.get_type(),
+            public_key_type: public_key.get_type(),
         });
         return Ok(validation_result);
     }
 
     let signature_is_valid = state_transition.verify_signature(public_key);
-    if signature_is_valid.is_err() {
-        validation_result.add_error(SignatureError::InvalidStateTransitionSignatureError);
+    // so we need to do is match the whole bunch of errors
+
+    if let Err(err) = signature_is_valid {
+        let consensus_error = convert_to_consensus_signature_error(err);
+        validation_result.add_error(consensus_error);
         return Ok(validation_result);
     }
 
     Ok(validation_result)
 }
 
+fn convert_to_consensus_signature_error(error: ProtocolError) -> ConsensusError {
+    match error {
+        ProtocolError::InvalidSignaturePublicKeySecurityLevelError {
+            public_key_security_level,
+            required_security_level,
+        } => ConsensusError::SignatureError(
+            SignatureError::InvalidSignaturePublicKeySecurityLevelError {
+                public_key_security_level,
+                required_key_security_level: required_security_level,
+            },
+        ),
+        ProtocolError::PublicKeySecurityLevelNotMetError {
+            public_key_security_level,
+            required_security_level,
+        } => ConsensusError::SignatureError(SignatureError::PublicKeySecurityLevelNotMetError {
+            public_key_security_level,
+            required_security_level,
+        }),
+        ProtocolError::PublicKeyIsDisabledError { public_key } => {
+            ConsensusError::SignatureError(SignatureError::PublicKeyIsDisabledError {
+                public_key_id: public_key.get_id(),
+            })
+        }
+        _ => ConsensusError::SignatureError(SignatureError::InvalidStateTransitionSignatureError),
```

Comment:
We can check in the code that there's no codepath leaded to some different ProtocolError here, but it really can be some unrelated error and we can loose some info here?

Reply:
You're right! I've added another match for errors coming from `anyhow`. They won't be returned as consensus errors but as protocol errors. Thanks!